Gem/Code/Source/Weapons/WeaponGathers.cpp

```diff
@@ -0,0 +1,115 @@
+        AZ_Assert(gatherParams.m_travelSpeed > 0.0f, "GatherEntitiesMultiSegment called with an invalid travel speed! This will fail, use the non-segmented gather path instead.");
+
+        ShotResult result = ShotResult::DoNotTerminate;
+
+        const AZ::Transform& startTransform = inOutActiveShot.m_initialTransform;
+        const AZ::Vector3 sweep = (inOutActiveShot.m_targetPosition - startTransform.GetTranslation()).GetNormalized();
+
+        // World gravity for our current location (making the currently safe assumption that it's constant over the duration of our trace)
+        //const AZ::Vector3& gravity = gatherParams.m_bulletDrop ? gNovaGame->GetNetworkPhysicalWorld().GetWorldGravity(a_InOutActiveShot.GetInitialTransform().m_Position) : AZ::Vector3::CreateZero();
+        const AZ::Vector3& gravity = AZ::Vector3::CreateZero();
+        const float segmentTickSize = deltaTime / bg_MultitraceNumTraceSegments; // Duration in seconds of each cast segment
+        const AZ::Vector3 segmentStepOffset = sweep * gatherParams.m_travelSpeed; // Displacement (disregarding gravity) of our bullet over one second
+        const float maxTravelDistanceSq = gatherParams.m_castDistance * gatherParams.m_castDistance;
+
+        // We're not doing any lift or drift computations due to the magnus effects a bullet is subject to
+        // Any such adjustments, estimates for how fast the bullet is spinning due to muzzle exit velocity and the rifling of the gun, air density, temperature, etc...
+
+        const HitMultiple hitMultiple = gatherParams.m_multiHit ? HitMultiple::Yes : HitMultiple::No;
+
+        float currSegmentStartTime = inOutActiveShot.m_lifetimeSeconds;
+        AZ::Vector3 currSegmentPosition = inOutActiveShot.m_initialTransform.GetTranslation() + (segmentStepOffset * currSegmentStartTime) + (gravity * 0.5f * currSegmentStartTime * currSegmentStartTime);
+        for (uint32_t segment = 0; segment < bg_MultitraceNumTraceSegments; ++segment)
+        {
+            float nextSegmentStartTime = currSegmentStartTime + segmentTickSize;
+            AZ::Vector3 travelDistance = (segmentStepOffset * nextSegmentStartTime); // Total distance our shot has traveled as of this cast, ignoring arc-length due to gravity
+            AZ::Vector3 nextSegmentPosition = inOutActiveShot.m_initialTransform.GetTranslation() + travelDistance + (gravity * 0.5f * nextSegmentStartTime * nextSegmentStartTime);
+
+            const AZ::Transform currSegTransform = AZ::Transform::CreateFromQuaternionAndTranslation(inOutActiveShot.m_initialTransform.GetRotation(), currSegmentPosition);
+            const AZ::Vector3   segSweep = nextSegmentPosition - currSegmentPosition;
+
+            IntersectFilter filter(currSegTransform, segSweep, HitStatic::Yes, HitDynamic::Yes, hitMultiple, filteredNetEntityIds);
+            //gNovaGame->GetNetworkPhysicalWorld().WorldIntersect(gatherParams.m_gatherShape, filter, a_OutResultList);
```

Comment:
This looks like it can be removed

Reply:
same, critical operation that requires porting to the new physics API